### PR TITLE
TST: interpolate: add a regression test for bisplev integer overflow

### LINKED ISSUE #18498: TST: interpolate overflow xslow tests (low priority)
This is quite low priority, but some `xslow`-marked tests similar to https://github.com/scipy/scipy/pull/17212#pullrequestreview-1435450128, adjusted to verify appropriate errors with i.e., `pytest.raises` for the various codepaths guarded in that PR could be useful long-term to avoid regression.

I suspect all of the tests would be cleaned up variations of the one I drafted there, once the other 2-3 segfault codepaths are studied/reached.

### PR DESCRIPTION
**Pull request by ev-br**
#### Reference issue

https://github.com/scipy/scipy/issues/18498
https://github.com/scipy/scipy/pull/17212

#### What does this implement/fix?

Add a regression test from https://github.com/scipy/scipy/pull/17212#pullrequestreview-1435450128, guarded by `@pytest.mark.xslow`.

Trying it locally, `$ SCIPY_XSLOW=1 python dev.py test -t scipy/interpolate/tests/test_fitpack.py::test_gh_1766`
passes locally, while without the env variable the test is skipped.

While at it, add `PyErr_NoMemory()` calls to convert RuntimeErrors into `MemoryErrors`. 

The test is constructed by @tylerjereddy ,  this is reflected in the commit authorship :-).

#### Additional information

The other place where gh-17212 checks for integer overflow seems to have some sort of a check at the python level, https://github.com/scipy/scipy/blob/main/scipy/interpolate/_fitpack_impl.py#L570
so I'd much rather merge in the test as is and move on.

### CODE REVIEW

File: scipy/interpolate/tests/test_fitpack.py [hunk 1]
```diff
@@ -451,6 +451,30 @@ def test_bisplev_integer_overflow():
...
+    # the size value here can either segfault
+    # or produce a MemoryError on main
+    tx_ty_size = 500000
+    tck[0] = np.arange(tx_ty_size)
+    tck[1] = np.arange(tx_ty_size) * 4
+    tt_0 = np.arange(50)
+    tt_1 = np.arange(50) * 3
+    with pytest.raises(MemoryError):
+        v1 = bisplev(tt_0, tt_1, tck, 1, 1)
+
```
Comment (j-bowhay) [inline scipy/interpolate/tests/test_fitpack.py:475]: To make the linter happy
```suggestion
        bisplev(tt_0, tt_1, tck, 1, 1)
```

**Review by j-bowhay — APPROVED** [review]
